### PR TITLE
add package py-spyder

### PR DESCRIPTION
**Pull request by healther**
requires:
- ~~py-rope~~
- ~~py-qtconsole~~
- ~~py-qtawesome~~
- ~~py-qtpy~~
- ~~py-pyzmq~~
- ~~py-chardet~~
- ~~py-numpydoc~~

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-spyder/package.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    depends_on('py-nbconvert',        type=('build', 'run'))
+    depends_on('py-sphinx',           type=('build', 'run'))
+    depends_on('py-flake8',           type=('build', 'run'))
+    depends_on('py-pylint',           type=('build', 'run'))
+    depends_on('py-psutil',           type=('build', 'run'))
+    depends_on('py-qtawesome@0.4.1:', type=('build', 'run'))
+    depends_on('py-qtpy@1.1.0:',      type=('build', 'run'))
+    depends_on('py-pyzmq',            type=('build', 'run'))
+    depends_on('py-chardet@2:',       type=('build', 'run'))
+    depends_on('py-numpydoc',         type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-spyder/package.py]: Might also want `setuptools`.

File: var/spack/repos/builtin/packages/py-spyder/package.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    homepage = "https://github.com/spyder-ide/spyder"
+    url      = "https://pypi.io/packages/source/s/spyder/spyder-3.1.3.tar.gz"
+
+    version('3.1.3', '4b9b7c8c3e6dc00001e6e98473473c36')
+    version('2.3.9', 'dd01e07a77123c128ff79ba57b97c1d7')
+
+    depends_on('py-rope@0.9.4:',      type=('build', 'run'), when='^python@:3')
+    # depends_on('py-rope_py3k',        type=('build', 'run'), when='^python@3:')
+    depends_on('py-jedi@0.9.0',       type=('build', 'run'))
+    # otherwise collision with py-flake8
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-spyder/package.py]: Yeah, I wish `py-flake8`/`py-pyflakes` and friends weren't so strict about their dependency versions.

File: var/spack/repos/builtin/packages/py-spyder/package.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    depends_on('py-pyflakes@1.2.3',   type=('build', 'run'))
+    depends_on('py-pygments@2.0:',    type=('build', 'run'))
+    depends_on('py-qtconsole@4.2.0:', type=('build', 'run'))
+    depends_on('py-nbconvert',        type=('build', 'run'))
+    depends_on('py-sphinx',           type=('build', 'run'))
+    depends_on('py-flake8',           type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-spyder/package.py]: Shouldn't this be `pep8`, a.k.a `py-pycodestyle`?

File: var/spack/repos/builtin/packages/py-spyder/package.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    depends_on('py-pylint',           type=('build', 'run'))
+    depends_on('py-psutil',           type=('build', 'run'))
+    depends_on('py-qtawesome@0.4.1:', type=('build', 'run'))
+    depends_on('py-qtpy@1.1.0:',      type=('build', 'run'))
+    depends_on('py-pyzmq',            type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-spyder/package.py]: Missing dependency on `py-pickleshare`.

File: var/spack/repos/builtin/packages/py-spyder/package.py (hunk 3 above)
Comment (healther) [inline var/spack/repos/builtin/packages/py-spyder/package.py]: I'm not sure, but as far as I have gathered, ```flake8``` and ```pycodestyle``` are different packages, I might be wrong though

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-spyder/package.py]: Yes, but the `setup.py` for `py-spyder` has:
```python
install_requires = [                                                            
    'rope_py3k' if PY3 else 'rope>=0.9.4',                                      
    'jedi==0.9.0',                                                              
    'pyflakes',                                                                 
    'pygments>=2.0',                                                            
    'qtconsole>=4.2.0',                                                         
    'nbconvert',                                                                
    'sphinx',                                                                   
    'pep8',                                                                     
    'pylint',                                                                   
    'psutil',                                                                   
    'qtawesome>=0.4.1',                                                         
    'qtpy>=1.1.0',                                                              
    'pickleshare',                                                              
    'pyzmq',                                                                    
    'chardet>=2.0.0',                                                           
    'numpydoc',                                                                 
] 
```
It lists `pep8`, not `flake8`.

File: var/spack/repos/builtin/packages/py-spyder/package.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-spyder/package.py]: This should be `py-zmq`. Lots of confusing duplicate packages.

**Comment by adamjstewart** [comment]
You'll have to let me know when all of the dependencies are merged. Then I can merge this package.

**Comment by healther** [comment]
We need #3311, #3307, #3308 to be merged before this can be merged

**Comment by adamjstewart** [comment]
Done. Thanks!

**Pull request merged**